test/cppsuite/tests/cursor_bound_02.cpp

```diff
@@ -0,0 +1,313 @@
+              connection_manager::instance().create_session(), tsm, op_tracker, database);
+            workers.push_back(tc);
+            tm.add_thread(populate_worker, tc, ALPHABET, PREFIX_KEY_LEN);
+        }
+
+        /* Wait for our populate threads to finish and then join them. */
+        logger::log_msg(LOG_INFO, "Populate: waiting for threads to complete.");
+        tm.join();
+
+        /* Cleanup our workers. */
+        for (auto &it : workers) {
+            delete it;
+            it = nullptr;
+        }
+
+        /* Force evict all the populated keys in all of the collections. */
+        int cmpp;
+        scoped_session session = connection_manager::instance().create_session();
+        for (uint64_t count = 0; count < collection_count; ++count) {
+            collection &coll = database.get_collection(count);
+            scoped_cursor evict_cursor =
+              session.open_scoped_cursor(coll.name.c_str(), "debug=(release_evict=true)");
+
+            for (uint64_t i = 0; i < ALPHABET.size(); ++i) {
+                for (uint64_t j = 0; j < ALPHABET.size(); ++j) {
+                    for (uint64_t k = 0; k < ALPHABET.size(); ++k) {
+                        std::string key = {ALPHABET.at(i), ALPHABET.at(j), ALPHABET.at(k)};
+                        evict_cursor->set_key(evict_cursor.get(), key.c_str());
+                        testutil_check(evict_cursor->search_near(evict_cursor.get(), &cmpp));
+                        testutil_check(evict_cursor->reset(evict_cursor.get()));
+                    }
+                }
+            }
+        }
+        srchkey_len =
+          random_generator::instance().generate_integer(static_cast<uint64_t>(1), PREFIX_KEY_LEN);
+        logger::log_msg(LOG_INFO, "Populate: finished.");
+    }
+
+    static void
+    perform_search_near(thread_worker *tc, std::string collection_name, uint64_t srchkey_len,
+      std::atomic<int64_t> &z_key_searches)
+    {
+        std::string srch_key;
+        int cmpp = 0;
+
+        scoped_cursor cursor = tc->session.open_scoped_cursor(collection_name);
+        /* Generate search prefix key of random length between a -> zzz. */
+        srch_key = random_generator::instance().generate_random_string(
+          srchkey_len, characters_type::ALPHABET);
+        logger::log_msg(LOG_TRACE,
+          "Search near thread {" + std::to_string(tc->id) +
+            "} performing prefix search near with key: " + srch_key);
+
+        /*
+         * Read at timestamp 10, so that no keys are visible to this transaction. When performing
+         * prefix search near, we expect the search to early exit out of its prefix range and return
+         * WT_NOTFOUND.
+         */
+        tc->txn.begin("read_timestamp=" + tc->tsm->decimal_to_hex(10));
+        if (tc->txn.active()) {
+            cursor->set_key(cursor.get(), srch_key.c_str());
+            bound_set prefix_bounds = bound_set(srch_key);
+            prefix_bounds.apply(cursor);
```

Comment:
Here I use the bound logic.

Reply:
Looks good but I kind of wish we would have something more natural like:
```
cursor->apply_bounds(bounds)
```

Reply:
I agree with this

Reply:
I am not a fan of giving the cursor knowledge about bounds.

Reply:
If you want to remove those bounds to that cursor, what do you do?

Reply:
You call reset on the cursor.